### DISCUSSION TITLE
VisiData GPT assistant: We just need a big text file

### DISCUSSION THREAD

**Discussion by cool-RR** [saulpw/visidata#2108] [Ideas]
Now that OpenAI allows people to create their own GPT, the dream of having an AI answering basic VisiData queries is closer than ever.

If anyone is interested in putting in the effort, here's what needs to be done: Assemble a big file (text or PDF) that has lots of information about VisiData. It should be one or two tutorials, and a list of questions and answers from the discussions, and the contents of the help menus, and maybe some history from the chat, if that can be programmatically separated to a question-answer format. It can also include discussions from pull requests. All the different kinds detailed above should be clearly marked so GPT knows in what context to take them.

If anyone makes a file like that, I can build the VisiData GPT. (It's pretty simple and you could do it yourself, but in any case I'll be happy to do it.)

**Comment by frosencrantz** [comment]
I think most answer/questions are here on GitHub, you can use https://github.com/dogsheep/github-to-sqlite .

Saul has many YouTube video tutorials, and there is a way to download the text transcript of the videos.

**Comment by saulpw** [comment]
Also there is now this, which can make a RAG from any sphinx-generated docsite (which ours is): https://github.com/jerpint/RAGTheDocs

**Comment by jgarte** [comment]
Is this still of interest to people? @cool-RR

**Reply by cool-RR** [reply]
I'd probably use it less than I would back when I asked this question, because now I'm better at VisiData, so my questions would be more advanced and challenging. I do think it's a good idea but I worry that it won't be discoverable enough to new users.

**Comment by aborruso** [comment]
Hi, a few weeks ago I developed a tool that answers questions about Visidata and other tools I love: Visidata, Miller, Xan, etc.

I haven't mentioned it yet because I wanted to get it right, but there's no point in wasting any more time. Here's how it works: type in the name of the tool and then ask your question.

For example: [VisiData: how do I create a pivot table?](https://chatgpt.com/share/68aff8d5-0e60-8002-a2dd-da19ddba3747)

This uses chatgpt, but it can be done in dozens of other ways. The main value is to generate a good text file to query.

The proposal to make an llmstxt file available to every site is very interesting: https://llmstxt.org/

For visidata, I created and used this file: https://gist.github.com/aborruso/590af90e7c04837e7807d52155d2cef2

I created it with repomix: https://repomix.com/guide/usage

I hope all this is useful.

**Reply by aborruso** [reply]
And there is also this great way to chat with visidata docs and code https://deepwiki.com/saulpw/visidata

**Reply by cool-RR** [reply]
Looks good , thank you

**Reply by frosencrantz** [reply]
Also, https://context7.com/saulpw/visidata
